### PR TITLE
Feature/fix translate default language

### PR DESCRIPTION
**Pull request by Sinisa198**

### CODE REVIEW

File: moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx [hunk 1]
```diff
@@ -1,12 +1,38 @@
...
     style,
 }: IErrorLabelProps): JSX.Element => {
-    return <label className={`errorLabel ${style}`}>{content}</label>
+    const { t } = useTranslation()
+    const [beforeEmail, setBeforeEmail] = useState<string>('')
+    const [email, setEmail] = useState<string>('')
+    const [afterEmail, setAfterEmail] = useState<string>('')
+
+    useEffect(() => {
+        const regex = /(.*) email (.*?) (.*?)$/
```
Comment (kiljee) [inline moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx:19]: regex on the top of file

File: moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx [hunk 2]
```diff
@@ -1,12 +1,38 @@
...
+        const matches = content?.match(regex)
+        if (matches && matches.length === 4) {
```
Comment (kiljee) [inline moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx:21]: matches?.length is same as matches && matches.length

File: moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx [hunk 3]
```diff
@@ -1,12 +1,38 @@
...
+            setBeforeEmail(matches[1])
```
Comment (kiljee) [inline moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx:22]: what is 1?

File: moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx [hunk 4]
```diff
@@ -1,12 +1,38 @@
...
+            setEmail(matches[2])
+            setAfterEmail(matches[3])
+        }
+    }, [content])
+
+    let errorMessage = ''
+    if (email) {
+        errorMessage = `${beforeEmail && t(beforeEmail)} ${email} ${
```
Comment (kiljee) [inline moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx:30]: you can write 
const errorMessage = error ? `` : ``

**Review by kiljee — CHANGES_REQUESTED** [review]

File: moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx [hunk 5]
```diff
@@ -1,12 +1,37 @@
...
+
 export const ErrorLabel = ({
     content,
     style,
 }: IErrorLabelProps): JSX.Element => {
-    return <label className={`errorLabel ${style}`}>{content}</label>
+    const { t } = useTranslation()
+    const [beforeEmail, setBeforeEmail] = useState<string>('')
+    const [email, setEmail] = useState<string>('')
+    const [afterEmail, setAfterEmail] = useState<string>('')
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/components/label/ErrorLabel.tsx:18]: move to single useState

File: moja-kuhinjica-frontend/src/components/modal/CanCancelReservation.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                </label>
+                <button
+                    className="modalContainer__formContainer__formDiv__formTitle__formButton"
+                    onClick={closeModal}
+                >
+                    {t(buttonText)}
+                </button>
+            </div>
+        )
+    }
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/components/modal/CanCancelReservation.tsx:42]: remove this function, use conditional rendering

File: moja-kuhinjica-frontend/src/components/modal/disabledReservation/DisabledReservationModal.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+                </label>
+                <button
+                    className="modalContainer__formContainer__formDiv__formTitle__formButton"
+                    onClick={closeModal}
+                >
+                    {t(buttonText)}
+                </button>
+            </div>
+        )
+    }
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/components/modal/disabledReservation/DisabledReservationModal.tsx:42]: remove this function, use conditional rendering

File: moja-kuhinjica-frontend/src/pages/email/password/[token].tsx [hunk 8]
```diff
@@ -8,8 +8,10 @@ import { useRouter } from 'next/router'
 import UserService from '@/service/User.service'
 import { Oval } from 'react-loader-spinner'
 import { routes } from '@/constants/constants'
+import { useTranslation } from 'react-i18next'
 
 const ResetPasswordPage = (): JSX.Element => {
+    const { t } = useTranslation()
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/email/password/[token].tsx:14]: use this `t` to translate

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 9]
```diff
@@ -40,11 +42,13 @@ import { routes } from '../../constants/constants'
...
 const HEADER_TYPE = 'red'
 const INITIAL_MEAL_AMOUNT = 1
+const DISABLED_MESSAGE = 'Ne možete da rezervišete posle 10 ujutru'
 const RESERVATION_SUCCESS = 'Rezervacija je uspešna'
 const RESERVATION_FAIL = 'Neuspešna rezervacija'
 const RESERVATION_SUCCESS_MESSAGE =
     'Vaša rezervacija je sačuvana. Možete je pogledati na stranici'
-const EMPTY_CART_MESSAGE = 'Vaša korpa je prazna, ukoliko želite da pogledate svoje rezervacije možete otići na stranicu'
+const EMPTY_CART_MESSAGE =
+    'Vaša korpa je prazna, ukoliko želite da pogledate svoje rezervacije možete otići na stranicu'
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:51]: move them to separate file so they don't clutter this one

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 10]
```diff
@@ -136,6 +142,10 @@ const MealReservation = (): JSX.Element => {
             })
     }
     const addToCart = (meal: IMeal): void => {
+        if (!isBookingAllowed(activeDate)) {
+            return setShowDisabledReservation(true)
+        }
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:147]: inline if

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 11]
```diff
@@ -237,6 +247,13 @@ const MealReservation = (): JSX.Element => {
                 }}
                 buttonText="OK"
             />
+            <DisabledReservationModal
+                modalIsOpen={showDisabledReservation}
+                closeModal={() => setShowDisabledReservation(!true)}
+                title={DISABLED_MESSAGE}
+                buttonText={'OK'}
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:254]: buttonText='OK'

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 12]
```diff
@@ -275,11 +292,11 @@ const MealReservation = (): JSX.Element => {
...
                                         onClick={() => {
-                                            setDeyOfWeek(date.day())
-                                            setActive(activeTabIndex)
                                             if (cartItems.length) {
                                                 handleTabClickWithCartItems()
                                             } else {
+                                                setDeyOfWeek(date.day())
+                                                setActive(activeTabIndex)
                                                 setActiveDate(day.date)
                                             }
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:301]: don't use `if - else`

File: moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx [hunk 13]
```diff
@@ -342,12 +360,17 @@ const MealReservation = (): JSX.Element => {
...
+                                        <span
                                             className="mealReservation__container__menuDiv__cartContainer__cartWrapper__emptyCartDiv__emptyCartLabel__emptySpan"
-                                            onClick={() => router.push(routes.MY_RESERVATIONS_PAGE)}    
-                                        > 
+                                            onClick={() =>
+                                                router.push(
+                                                    routes.MY_RESERVATIONS_PAGE
+                                                )
+                                            }
+                                        >
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/mealReservation/index.tsx:373]: use Link

File: moja-kuhinjica-frontend/src/pages/myReservations/index.tsx [hunk 14]
```diff
@@ -29,26 +29,14 @@ import mealDefault from 'public/static/assets/images/mealDefault.svg'
...
-                isActive
-                onClick={() => router.push(routes.MEAL_RESERVATION_PAGE)}
-            />
-        </div>
-    )
-}
+const NoReservationsMessage: React.FC = () => (
+    <div className="myReservationsPage__colDiv__rowDiv">
+        <label className="myReservationsPage__colDiv_an_rowDiv__infoLabel">
+            Nema rezervacija za ovaj datum.
```
Comment (vladavoX) [inline moja-kuhinjica-frontend/src/pages/myReservations/index.tsx:36]: translate this

**Review by vladavoX — APPROVED** [review]

**Review by kiljee — APPROVED** [review]

**Pull request merged**